lib/ui/src/components/notifications/item.tsx

```diff
@@ -1,34 +1,141 @@
+  },
+  '&:active': {
+    boxShadow:
+      '0 1px 3px 0 rgba(30,167,253,0.5), 0 2px 5px 0 rgba(0,0,0,0.05), 0 5px 15px 0 rgba(0,0,0,0.1)',
+  },
+  '&:focus': {
+    boxShadow:
+      '0 1px 3px 0 rgba(30,167,253,0.5), 0 2px 5px 0 rgba(0,0,0,0.05), 0 5px 15px 0 rgba(0,0,0,0.1)',
+  },
 }));
+
 const NotificationLink = Notification.withComponent(Link);
 
+const NotificationIconWrapper = styled.div(() => ({
+  display: 'flex',
+  marginRight: 10,
+  alignItems: 'center',
+}));
+
+const NotificationTextWrapper = styled.div(() => ({
+  width: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+}));
+
+const Headline = styled.div(({ theme }) => ({
+  display: 'flex',
+  height: '100%',
+  alignItems: 'center',
+  color: theme.base === 'light' ? '#fff' : '#333333',
+  fontSize: 12,
+  lineHeight: '14px',
+  fontWeight: 'bold',
+}));
+
+const SubHeadline = styled.div(({ theme }) => ({
+  color: theme.base === 'light' ? 'rgba(255,255,255,0.7)' : ' #999999',
+  fontSize: 11,
+  fontWeight: 'normal',
+  marginTop: 3,
+  marginRight: 5,
+  lineHeight: 1,
+}));
+
+const truncateLongHeadlines = (headline: string, length = 30) =>
+  headline.length > length ? `${headline.substr(0, length - 1)}…` : headline;
+
+const ItemContent: FunctionComponent<Pick<State['notifications'][0], 'icon' | 'content'>> = ({
+  icon,
+  content: { headline, subHeadline },
+}) => (
+  <>
+    {!icon || (
+      <NotificationIconWrapper>
+        <Icons
+          icon={icon.name as IconsProps['icon']}
+          width={14}
+          color={icon.color || DEFAULT_ICON_COLOUR}
+        />
+      </NotificationIconWrapper>
+    )}
+    <NotificationTextWrapper>
+      <Headline title={headline}>{truncateLongHeadlines(headline)}</Headline>
+      {subHeadline && <SubHeadline>{subHeadline}</SubHeadline>}
+    </NotificationTextWrapper>
+  </>
+);
+
+const DismissButtonWrapper = styled(IconButton)(({ theme }) => ({
+  height: '100%',
+  display: 'flex',
+  color: theme.base === 'light' ? 'rgba(255,255,255,0.7)' : ' #999999',
+}));
+
+const DismissNotificationItem: FunctionComponent<{
+  onClick: () => void;
+}> = ({ onClick }) => (
+  <div>
+    <DismissButtonWrapper
+      title="Dismiss notification"
+      onClick={(e: SyntheticEvent) => {
+        e.preventDefault();
+        onClick();
+      }}
+    >
+      <Icons icon="closeAlt" height={12} width={12} />
+    </DismissButtonWrapper>
+  </div>
+);
+
 export const NotificationItemSpacer = styled.div({
   height: 48,
 });
 
 const NotificationItem: FunctionComponent<{
   notification: State['notifications'][0];
-}> = ({ notification: { content, link } }) => {
+  setDismissedNotification: (id: string) => void;
+}> = ({ notification: { content, link, onClear, id, icon }, setDismissedNotification }) => {
+  const dismissNotificationItem = () => {
+    setDismissedNotification(id);
+    onClear();
+  };
   return link ? (
-    <NotificationLink to={link}>{content}</NotificationLink>
+    <NotificationLink to={link}>
```

Comment:
When there is a link present the NotificationItem should have a hover effect. Right now the hover effect is blanket applied to all NotificationItems even if they aren't interactive.